### PR TITLE
Test Controllers

### PR DESCRIPTION
**Pull request by yonas44**
## These are the changes implemented in this PR:
- Create a Request spec file for all your controllers.
- Make sure that for each action you check: 
- [x] If response status was correct.
- [x] If a correct template was rendered.
- [x] If the response body includes correct placeholder text.

### CODE REVIEW

File: spec/controllers/posts_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
+require 'rails_helper'
+
+RSpec.describe 'Posts', type: :request do
+  describe 'GET #index' do
+    before(:example) { get '/users/1/posts' }
+    it 'returns successful' do
+      expect(response).to have_http_status(:ok)
+    end
+    it 'assigns an instance variable accessable from the view' do
```
Comment (Ismailco) [inline spec/controllers/posts_controller_spec.rb:9]: - [ ] Kindly make sure that your Spec file includes tests that ensure the response body includes the correct placeholder text, this should be included in the tests for `index` and `show` templates/pages
e.g
```ruby
expect(response.body).to include("Welcome to user index page")
```

File: spec/controllers/users_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+RSpec.describe 'Users', type: :request do
+  describe 'GET #index' do
+    before(:example) { get users_path }
+    it 'is a success' do
+      expect(response).to have_http_status(:ok)
+    end
+    it "renders 'index' template" do
+      expect(response).to render_template('index')
+    end
+    it 'assigns a variable to be accessed in views' do
```
Comment (Ismailco) [inline spec/controllers/users_controller_spec.rb:12]: - [ ] Kindly make sure that your Spec file includes tests that ensure the response body includes the correct placeholder text, this should be included in the tests for `index` and `show` templates/pages
e.g
```ruby
expect(response.body).to include("Welcome to user index page")
```

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @yonas44, 

Good job so far 👍🏼 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ No linter errors
✔️ Include rspec tests
✔️ Use Gitflow
✔️ Work documented professionally

## Required Changes ♻️
- [x] Kindly make sure to add the request test files into the `spec/requests` folder, because these tests are not for `controllers`

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/controllers/users_controller_spec.rb (hunk 2 above)
Comment (yonas44) [reply spec/controllers/users_controller_spec.rb:12]: Okay, thanks for the feedback, I will address them all.

**Review by Omar-Muhamad — APPROVED** [review]
# Approved ✔
### Hi @yonas44,

Great job, Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

thumbs-up-agree


## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used the correct workflow.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- N/A

## [Optional] suggestions:

- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**